### PR TITLE
refactor: added pacote to resolve template name instead of arborist workaround

### LINKED ISSUE #1102: Improve arborist (npm installation) to have no hacks
use [this PR](https://github.com/asyncapi/generator/pull/925/files) as inspiration, except for:
- no changes to workflows
- arborist version should not be changed by this issue
- supported engines should not be changed by this issue

What you need to do:
- just ad `pacote` dependency
- cleanup mocks and add new one - like in linked PR
- modify `generator.js` as it is done in linked PR

### PR DESCRIPTION
**Pull request by ShashwatPS**
### **Description**

- Added `pacote` dependency.
- Cleaned `arbortis` mock and added a new one for `pacote`.
- Modified `generator.js` as suggested.

### **Related issue(s)**

- Resolves #1102

### CODE REVIEW

File: apps/generator/lib/generator.js [hunk 1]
```diff
@@ -605,10 +606,8 @@ class Generator {
         save: false
       });
 
-      const addResult = arb[Symbol.for('resolvedAdd')];
-      if (!addResult) throw new Error('Unable to resolve the name of the added package. It was most probably not added to node_modules successfully');
-
-      const packageName = addResult[0].name;
+      const manifest = await pacote.manifest(this.templateName);
```
Comment (DhairyaMajmudar) [inline apps/generator/lib/generator.js:609]: Here how errors are handled? If they are not pls. add conditions for that

File: apps/generator/test/test-project/verdaccio/config.yaml
Comment (DhairyaMajmudar) [inline apps/generator/test/test-project/verdaccio/config.yaml:1]: Pls. remove empty file if its not required

File: apps/generator/test/test-project/verdaccio/htpasswd
Comment (DhairyaMajmudar) [inline apps/generator/test/test-project/verdaccio/htpasswd:1]: Remove this empty file also

File: apps/generator/test/test-templates/react-template/__transpiled/test-file.md.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 require('source-map-support/register');
-const generatorReactSdk = require('@asyncapi/generator-react-sdk');
-const jsxRuntime = require('react/cjs/react-jsx-runtime.production.min');
+var generatorReactSdk = require('@asyncapi/generator-react-sdk');
+var jsxRuntime = require('C:/Users/shash/Desktop/generator/node_modules/react/cjs/react-jsx-runtime.production.min.js');
```
Comment (DhairyaMajmudar) [inline apps/generator/test/test-templates/react-template/__transpiled/test-file.md.js:5]: Here the file path is compatible to windows OS, if someone is using Linux or Mac. Won't it fail to execute?

**Comment by derberg** [comment]
@ShashwatPS do you plan to followup with initial review from @DhairyaMajmudar ?

**Pull request closed without merging**